src/services/textChanges.ts

```diff
@@ -1296,6 +1305,11 @@ namespace ts.textChanges {
                     deleteNode(changes, sourceFile, node, { leadingTriviaOption: LeadingTriviaOption.Exclude });
                     break;
 
+                case SyntaxKind.ClassDeclaration:
+                case SyntaxKind.FunctionDeclaration:
+                    deleteNode(changes, sourceFile, node, { leadingTriviaOption: hasJSDocNodes(node) ? LeadingTriviaOption.IncludeAll : LeadingTriviaOption.StartLineOnly });

```

Comment:
The conditional between `IncludeAll` and `StartLineOnly` makes me think you might need to take a look at cases like this:

```ts
// don't want to delete this comment

/** @param p delete this comment */
function unusedFunction(p) { }
```

Reply:
Good catch - I'm thinking about this wrong.

Reply:
I added another leading trivia option `JSDoc` to remove all JSDoc comments which belong to a node.

```ts
// singleLine1
/**
* JSDoc1
*/
// singleLine2
/**
* JSDoc2
*/
function f(){}
```

now becomes

```ts|
// singleLine1
```

Seems like removing `singleLine2` is appropriate if we remove all JSDoc comments belonging to that node. Otherwise, only deleting `JSDoc2` (and leaving `JSDoc2`) also seems sane...

Reply:
Yeah, the weirder the scenario, the more approaches there are that seem reasonable. If in doubt I’d go conservative and delete less, but 🤷‍♂